Quadriga/src/main/java/edu/asu/spring/quadriga/utilities/impl/TextXMLParser.java

```diff
@@ -0,0 +1,71 @@
+package edu.asu.spring.quadriga.utilities.impl;
+
+import java.io.IOException;
+import java.io.StringReader;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import edu.asu.spring.quadriga.domain.factory.workspace.ITextFileFactory;
+import edu.asu.spring.quadriga.domain.workspace.ITextFile;
+import edu.asu.spring.quadriga.exceptions.TextFileParseException;
+import edu.asu.spring.quadriga.utilities.ITextXMLParser;
+
+@Service
+public class TextXMLParser implements ITextXMLParser {
```

Comment:
this class should work hand in hand with NetworkXMLParser. Make sure that you don't duplicate code. I guess it would make the most sense if the NetworkXMLParser would call this class to create the text file.